### ISSUE TITLE
Union operator (`|`) incompatible with __future__.annotations

### ISSUE THREAD

**Issue by floringogianu** [roee30/datargs#34]
It appears the parser breaks when using the `|` operator in conjunction with importing `annotations` from `__futures__`. On `python 3.12` with `datargs` installed from `main` this example errors out:

```python
from __future__ import annotations

from datargs import arg, argsclass, parse


@argsclass
class Eat:
    food: str = arg(positional=True)

    def run(self):
        print("eating", self.food)


@argsclass
class Sleep:
    time: str = arg(positional=True)

    def run(self):
        print("sleeping", self.time)


@argsclass
class App:
    action: Eat | Sleep

    def run(self):
        self.action.run()


if __name__ == "__main__":
    args = parse(App)
    args.run()

```

```shell
Traceback (most recent call last):
...
  File ".../lib/python3.12/site-packages/datargs/make.py", line 481, in parse
    result = vars(make_parser(cls, parser=parser).parse_args(args))
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../lib/python3.12/site-packages/datargs/make.py", line 372, in make_parser
    return _make_parser(record_class, parser=parser)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../lib/python3.12/site-packages/datargs/make.py", line 420, in _make_parser
    parser.add_argument(*action.args, **action.kwargs)
  File ".../lib/python3.12/argparse.py", line 1477, in add_argument
    raise ValueError('%r is not callable' % (type_func,))
ValueError: 'Eat | Sleep' is not callable
```

It works just fine if you don't import from `__futures__`.

**Comment by roee30** [comment]
Published fixed in 1.1.0

**Comment by floringogianu** [comment]
Thank you!
